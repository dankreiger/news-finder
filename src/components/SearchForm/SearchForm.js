import React from 'react';
import { func, string } from 'prop-types';
import { Col, Row } from 'reactstrap';
import SearchButton from '../SearchButton/SearchButton';
import SearchInput from '../SearchInput/SearchInput';

const SearchForm = ({ handleChange, fetchArticles, inputVal }) => (
  <Row>
    <Col xs="12" sm="9" md="10">
      <SearchInput
        handleChange={handleChange}
        inputVal={inputVal}
        placeholder="Enter a search term"
      />
    </Col>
    <Col xs="12" sm="3" md="2">
      <SearchButton handleClick={fetchArticles} theme="primary" />
    </Col>
  </Row>
);

SearchForm.propTypes = {
  fetchArticles: func.isRequired,
  handleChange: func.isRequired,
  inputVal: string
};

export default SearchForm;
