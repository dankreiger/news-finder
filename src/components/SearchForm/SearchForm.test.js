import React from 'react';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('App', () => {
  let component;
  const fetchArticles = jest.fn();
  const handleChange = jest.fn();

  beforeEach(() => {
    component = shallow(
      <SearchForm fetchArticles={fetchArticles} handleChange={handleChange} />
    );
  });

  it('renders without crashing', () => {
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('renders an SearchInput component', () => {
    expect(component.find('SearchInput').length).toBe(1);
  });

  it('renders an SearchButton component', () => {
    expect(component.find('SearchButton').length).toBe(1);
  });
});
