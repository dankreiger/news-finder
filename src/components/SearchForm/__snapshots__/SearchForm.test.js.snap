// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    fetchArticles={[MockFunction]}
    handleChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          md="10"
          sm="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <SearchInput
            handleChange={[MockFunction]}
            placeholder="Enter a search term"
          />
        </Col>,
        <Col
          md="2"
          sm="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <SearchButton
            handleClick={[MockFunction]}
            text="Click Me"
            theme="primary"
          />
        </Col>,
      ],
      "className": "mb-3",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <SearchInput
            handleChange={[MockFunction]}
            placeholder="Enter a search term"
          />,
          "md": "10",
          "sm": "9",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [MockFunction],
            "inputVal": undefined,
            "placeholder": "Enter a search term",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <SearchButton
            handleClick={[MockFunction]}
            text="Click Me"
            theme="primary"
          />,
          "md": "2",
          "sm": "3",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [MockFunction],
            "text": "Click Me",
            "theme": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            md="10"
            sm="9"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <SearchInput
              handleChange={[MockFunction]}
              placeholder="Enter a search term"
            />
          </Col>,
          <Col
            md="2"
            sm="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <SearchButton
              handleClick={[MockFunction]}
              text="Click Me"
              theme="primary"
            />
          </Col>,
        ],
        "className": "mb-3",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <SearchInput
              handleChange={[MockFunction]}
              placeholder="Enter a search term"
            />,
            "md": "10",
            "sm": "9",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": [MockFunction],
              "inputVal": undefined,
              "placeholder": "Enter a search term",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <SearchButton
              handleClick={[MockFunction]}
              text="Click Me"
              theme="primary"
            />,
            "md": "2",
            "sm": "3",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleClick": [MockFunction],
              "text": "Click Me",
              "theme": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
